src/group_elements.jl

```diff
@@ -1,223 +1,264 @@
 """
-Base.:(*)(g::GEl, h::GEl) where {GEl<:GroupElement} = throw(
+Base.:(*)(g::G, h::G) where {G <: GroupElement} = throw(
     InterfaceNotImplemented(
         :Group,
         "Base.:(*)(::$(typeof(g)), ::$(typeof(g)))",
     ),
 )
 
-### Default implementations for `GroupElement`
+################################################################################
+# Default implementations
+################################################################################
 
-#### Modification not recommended
 @doc Markdown.doc"""
     one(g::GroupElement)
-Return the identity element of the parent group of `g`.
+
+Return the identity element in the group of $g$.
 """
 Base.one(g::GroupElement) = one(parent(g))
 
 @doc Markdown.doc"""
-    order([BigInt, ]g::GroupElement)
-    order(I::Type{<:Integer}, g::GroupElement)
-Return the order of `g` as an instance of `I`.
+    order(::Type{I} = BigInt, g::GroupElement) where {I <: Integer}
+
+Return the order of $g$ as an instance of `I`. If $g$ is of infinite order
+`GroupsCore.InfiniteOrder` exception will be thrown.
 
-Only arbitrary size integers are required to return mathematically correct answer.
+!!! warning
+
+    Only arbitrary sized integers are required to return a mathematically
+    correct answer.
 """
-AbstractAlgebra.order(g::GroupElement) = order(BigInt, g)
+function order(::Type{I}, g::GroupElement) where {I<:Integer}
+    isfiniteorder(g) || throw(InfiniteOrder(g))
+    isone(g) && return I(1)
+    o = I(1)
+    gg = deepcopy(g)
+    out = similar(g)
+    while !isone(gg)
+        o += I(1)
+        gg = mul!(out, gg, g)
+    end
+    return o
+end
+order(g::GroupElement) = order(BigInt, g)
 
 @doc Markdown.doc"""
-    conj(g::GEl, h::GEl) where {GEl<:GroupElement}
-Return conjugation of `g` by `h`, i.e. `h^-1*g*h`.
+    conj(g::G, h::G) where {G <: GroupElement}
+
+Return conjugation of $g$ by $h$, i.e. $h^{-1} g h$.
 """
-Base.conj(g::GEl, h::GEl) where {GEl<:GroupElement} = conj!(similar(g), g, h)
+Base.conj(g::G, h::G) where {G <: GroupElement} = conj!(similar(g), g, h)
 
 @doc Markdown.doc"""
-    ^(g::GEl, h::GEl) where {GEl<:GroupElement}
-Conjugation action of `h` on `g`. See `conj`.
+    ^(g::G, h::G) where {G <: GroupElement}
+
+Alias for [`conj`](@ref GroupsCore.conj).
 """
-Base.:(^)(g::GEl, h::GEl) where {GEl<:GroupElement} = conj(g, h)
+Base.:(^)(g::G, h::G) where {G <: GroupElement} = conj(g, h)
 
 @doc Markdown.doc"""
-    comm(g::GEl, h::GEl[, Vararg{GEl}...) where GEl<:GroupElement
-Return the commutator `g^-1*h^-1*g*h` of `g` and `h`.
+    comm(g::G, h::G, k::G...) where {G <: GroupElement}
 
-The `Vararg` version returns the repeated (`foldl`) commutator, i.e.
-`comm(g, h, k) == comm(comm(g, h), k)`.
+Return the left associative iterated commutator $[[g, h], ...]$, where
+$[g, h] = g^{-1} h^{-1} g h$.
 """
-function comm(g::GEl, h::GEl, tail::GEl...) where {GEl<:GroupElement}
+function comm(g::G, h::G, k::G...) where {G <: GroupElement}
```

Comment:
Offtopic, but: I am not happy about the use of G for a type. It nay be just me, but G in the literature always is a group, and I am concerned it will lead to misunderstandings when reading the code, esp. by beginners. Even just plain `T` would IMHO be preferable

Reply:
yes, I share this opinion; I'd be happiest to use `G::Gr ... where Gr<:Group` and `g::GEl ... where GEl<:GroupElement`. It's actually very rare that the type can function as a group object itself.

Reply:
I'll fix according to @kalmarek